vision/pipeline.py

```diff
@@ -212,47 +224,47 @@ def run(self, prev_state):
                                             :,
                                         ]
                                     )  # roll of module
-                                except:
-                                    flags.get_module_roll = False
-
-                                if flags.get_module_roll:
                                     # construct boundingbox for the module
                                     box = BoundingBox(bounds, ObjectType.MODULE)
                                     box.module_depth = depth  # float
                                     box.orientation = orientation + (
                                         roll,
                                     )  # x, y, z tilt
-
                                     bboxes.append(box)
+                                except:
+                                    flags.get_module_roll = False
 
         else:
             pass  # raise AttributeError(f"Unrecognized state: {state}")
 
-        self.vision_flags.append(flags)
+        time = datetime.datetime.now()
 
+        # You need to index vision_flags to see output, "self.vision_flags.get()[1]"
```

Comment:
Something I though of: If you change the `__str__()` functions in the flag classes to `__repr__()`, it might fix this so that the tuple doesn't need to be indexed. Should be as simple as changing the function names if you could test this. Classes are in `failure_flags.py`